packages/docs/src/content/docs/reference/faq.md

```diff
@@ -0,0 +1,326 @@
+Webpack loaders or Vite asset imports. Knip strips off the prefixes and suffixes
+in import specifiers like this:
+
+```ts title="component.ts"
+import Icon from './icon.svg?raw';
+import Styles from '-!style-loader!css-loader?modules!./styles.css';
+```
+
+In this example, the `style-loader` and `css-loader` dependencies should be
+dependencies found in Webpack configuration, handled by Knip's Webpack plugin.
+
+## What does Knip look for in source files?
+
+The TypeScript source file parser is powerful and fault-tolerant. Knip visits
+all nodes of the generated AST to find:
+
+- Exports
+- Imports (incl. dynamic import calls) of internal modules and external
+  dependencies
+- Accessed properties on namespace imports and re-exports to track individual
+  export usage
+- Calls to `require.resolve` and `import.meta.resolve`
+- Scripts in template strings
+
+## Where does Knip look for entry files?
+
+- In [default locations][5] such as `index.js` and `src/index.ts`
+- In `main`, `bin` and `exports` fields in `package.json`
+- In the entry files as configured by enabled plugins
+- In `config` files as configured and parsed by enabled plugins
+- In dynamic imports (i.e. `require()` and `import()` calls)
+- In `require.resolve('./entry.js')`
+- In `import.meta.resolve('./entry.mjs')`
+- Through scripts inside template strings in source files such as:
+  ```ts
+  await $({ stdio: 'inherit' })`c8 node hydrate.js`; // execa
+  await $`node scripts/parse.js`; // bun/zx
+  ```
+- Through scripts in `package.json` such as:
+  ```json
+  {
+    "name": "my-lib",
+    "scripts": {
+      "start": "node --import tsx/esm run.ts"
+    }
+  }
+  ```
+
+Entry files are added to the module graph and they might lead to additional
+entry files recursively until no more entry files are found.
+
+## Why does Knip have plugins?
+
+Plugins are an essential part of Knip. They save a lot of configuration out of
+the box by adding entry files as accurately as possible and only for the tools
+actually installed. Yet the real magic is in their custom parsers for
+configuration files.
+
+For instance, Vitest has the `environment` configuration option. The Vitest
+plugin knows `"node"` is the default value for `environment` which does not
+require an extra package, but will translate `"edge-runtime"` to the
+`@edge-runtime/vm` package. This allows Knip to report it if this package is not
+listed in `package.json`, or when it is no longer used after changes in the
+Vitest configuration.
+
+Configuration files may also contain references to entry files. For instance,
+Jest has `setupFilesAfterEnv: "<rootDir>/jest.setup.js"` or a reference may
+point to a file in another workspace in the same monorepo, e.g.
+`setupFiles: ['@org/shared/jest-setup.ts']`. Those entry files may also contain
+imports of internal modules or external dependencies, and so on.
+
+With an easy-to-use plugin API, many plugins have been created by contributors.
```

Comment:
It's very polite of you to not say it's because the JS/TS ecosystem has plenty of tools that each have their own configuration files, source files, standard and non-standard syntax, and reference magic :smile: 

I think it's probably worth mentioning that the variety of approaches in the ecosystem does create this need for plugins. It also relates to the next question in that it impacts the engineering of Knip.

Reply:
I was trying to address this in the introduction of this page, but indeed could be emphasized better I guess.